src/main/java/com/couchbase/intellij/tools/DDLExport.java

```diff
@@ -0,0 +1,118 @@
+import com.couchbase.intellij.workbench.Log;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.ui.Messages;
+import org.jetbrains.annotations.NotNull;
+import utils.IndexUtils;
+import utils.TimeUtils;
+
+import javax.swing.*;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DDLExport {
+
+    private static String exportScope(String bucket, String scope, boolean includeIndexes) {
+        StringBuilder sb = new StringBuilder();
+        String scopePrefix = "`" + bucket + "`.`" + scope + "`";
+
+        if (!"_default".equals(scope)) {
+            sb.append("CREATE SCOPE ").append(scopePrefix).append(";");
+        }
+        sb.append("\n");
+
+        final List<CollectionSpec> collections = ActiveCluster.getInstance().get().bucket(bucket)
+                .collections().getAllScopes().stream()
+                .filter(s -> s.name().equals(scope))
+                .flatMap(s -> s.collections().stream())
+                .collect(Collectors.toList());
+
+        for (CollectionSpec spec : collections) {
+
+            sb.append("\n");
+            sb.append("/* DDL for collection ").append(scope).append(".").append(spec.name()).append(" */");
+            sb.append("\n");
+            if (!"_default".equals(spec.name())) {
+                sb.append("CREATE COLLECTION ").append(scopePrefix).append(".`").append(spec.name()).append("`; \n");
+            }
+            if (includeIndexes) {
+                List<QueryIndex> result = DataLoader.listIndexes(bucket, scope, spec.name());
+                sb.append(result.stream().map(IndexUtils::getIndexDefinition).collect(Collectors.joining("; \n")));
+
+                if (result.size() > 1) {
+                    sb.append("; \n");
+                }
+            }
+            sb.append("\n");
+        }
+
+        sb.append("\n\n");
+
+        return sb.toString();
+    }
+
+    public static void exportScope(final String bucket, final List<String> scopes, final String filePath, final boolean includeIndexes) {
+        ProgressManager.getInstance().run(new Task.Backgroundable(null, "Exporting DDL'", true) {
+            public void run(@NotNull ProgressIndicator indicator) {
+                indicator.setIndeterminate(true);
+                try {
+                    StringBuilder sb = new StringBuilder();
+
+                    boolean allScopes = scopes.contains("All Scopes");
+                    List<String> selectedScopes = scopes;
+
+                    if (allScopes) {
+                        selectedScopes = ActiveCluster.getInstance().get().bucket(bucket).collections().getAllScopes().stream().map(ScopeSpec::name).collect(Collectors.toList());
+                    }
+
+                    for (String scope : selectedScopes) {
```

Comment:
You could save a stream object instead of collecting everything into a list and then iterating over that list, for example:
`
Stream<String> selectedScopes = allScopes ? scopes.stream() : ActiveCluster...
`
and then `selectedScopes.map(...).collect(...)`